nvtabular/ops/hashed_cross.py

```diff
@@ -0,0 +1,57 @@
+#
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import cudf
+from nvtx import annotate
+
+from .operator import CAT
+from .transform_operator import TransformOperator
+
+
+class HashedCross(TransformOperator):
+    """"""
+
+    default_in = CAT
+    default_out = CAT
+
+    def __init__(self, crosses, num_buckets=None, **kwargs):
+        if isinstance(crosses, dict):
+            cross_sets = list(crosses.keys())
+            num_buckets = [crosses[c] for c in cross_sets]
+            crosses = cross_sets
+        else:
+            if num_buckets is None:
+                raise ValueError("Must provide `num_buckets` if crosses is not dict")
+            assert len(num_buckets) == len(crosses)
+        assert all([isinstance(c, (tuple, list)) for c in crosses])
+        assert all([len(c) > 1 for c in crosses])
+
+        kwargs["replace"] = False
+        kwargs["columns"] = list(set([i for j in crosses for i in j]))
+        super().__init__(**kwargs)
+
+        self.crosses = crosses
+        self.num_buckets = num_buckets
+
+    @annotate("HashedCross_op", color="darkgreen", domain="nvt_python")
+    def op_logic(self, gdf: cudf.DataFrame, target_columns: list, stats_context=None):
+        new_gdf = cudf.DataFrame()
+        for columns, bucket_size in zip(self.crosses, self.num_buckets):
+            val = 0
+            for column in columns:
+                val ^= gdf[column].hash_values()  # or however we want to do this aggregation
```

Comment:
Our categorify op lets you pass in column groups, and takes an 'encode_type' parameter - which if set to 'combo' does the categorical encoding on the cross like this op:

https://github.com/NVIDIA/NVTabular/blob/f39e65e95d0af1d44ae9c2073a06c5a442d4de93/nvtabular/ops/categorify.py#L76-L81  

What do you think about rolling the functionality for this op into the HashBucket op to be consistent? (like if passed a multi column group, then doing the cross). 

Ronay is working on merging the HashBucket functionality with the categorify op - and this is one of the things that I think we could do to minimize the delta in functionality between the two

Reply:
Hmmm that's an interesting idea... I think that could make sense. I'll need to think more about it from a conceptual standpoint but they definitely seem equivalent

Reply:
Based on the meeting today, do we want to shelve this until after the 0.3 release? While I agree that this functionality should all be wrapped together, the API standpoint feels non-trivial and at this point it might make more sense just to get this in for TF users (especially if it get primarily used by the `make_feature_column_workflow` function, since any API changes will be handled on the backend and wouldn't require users to update their code)

Reply:
sounds good - we can always come back to this